app/javascript/articles/components/SaveButton.jsx

```diff
@@ -20,25 +39,38 @@ export class SaveButton extends Component {
       }
     };
 
+    let printCollectionTitles = this.state.allCollectionTitles.map(title => {
+      return <option value={title}>{title}</option>
+    })
+
     if (article.class_name === 'Article') {
       return (
-        <button
-          type="button"
-          className={`article-engagement-count engage-button bookmark-button ${
-            isBookmarked ? 'selected' : ''
-          }`}
-          data-initial-feed
-          data-reactable-id={article.id}
-          onClick={onClick}
-          onMouseOver={mouseOver}
-          onFocus={mouseOver}
-          onMouseout={mouseOut}
-          onBlur={mouseOut}
-        >
-          <span className={isBookmarked ? 'bm-success' : 'bm-initial'}>
-            {buttonText}
-          </span>
-        </button>
+        <div className='save-to-collection-button-container'>
+            <div className='collection-dropdown-container'>
+              <label className='collection-dropdown-heading'>collection:</label>
+              <select className='collection-dropbox' onChange={e => this.updateDropboxState(e.target.value)}>
+                {printCollectionTitles}
+              </select>
+            </div>
+          <button
+            type="button"
+            className={`article-engagement-count engage-button bookmark-button ${
+              isBookmarked ? 'selected' : ''
+            }`}
+            data-initial-feed
+            data-collection={this.state.collectionDropboxValue}
```

Comment:
This was another great idea to leverage the data attribute of this component's HTML.